gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/s3_signer.rb

```diff
@@ -52,7 +52,10 @@ def call(context)
 
           def check_for_cached_region(context, bucket)
             cached_region = S3::BUCKET_REGIONS[bucket]
-            if cached_region && cached_region != context.config.region
+
+            if cached_region &&
+               cached_region != context.config.region &&
+               context[:endpoint_params].endpoint.nil?
```

Comment:
This is a little bit of a confusing internal detail - but we should use `context.config.regional_endpoint` instead of `context[:endpoint_params].endpoint.nil?`

`regional_endpoint` is set to `false` when a custom endpoint has been configured.